File: chapter3/app/src/main/java/com/example/chapter3/homework/MyAdapter.java
package com.example.chapter3.homework;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

public class MyAdapter extends RecyclerView.Adapter<MyAdapter.ViewHolder> {

    private List<String> list;

    static class ViewHolder extends RecyclerView.ViewHolder {
        TextView content;

        public ViewHolder(View view) {
            super(view);
            content = (TextView) view.findViewById(R.id.text_view);
        }

    }

    public MyAdapter(List<String> contentList) {
        this.list = contentList;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_layout, parent, false);
        ViewHolder holder = new ViewHolder(view);
        return holder;
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        holder.content.setText(list.get(position));
    }

    @Override
    public int getItemCount() {
        return list.size();
    }
}

File: chapter1/app/src/main/java/com/example/searchanddisplay/MyAdapter.kt
package com.example.searchanddisplay

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView


class MyAdapter(val contentList: List<String>) : RecyclerView.Adapter<MyViewHolder>() {
    private lateinit var mContext: Context

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        mContext = parent.context
        val itemView =
            LayoutInflater.from(parent.context).inflate(R.layout.item_layout, parent, false)
        return MyViewHolder(itemView)
    }

    override fun getItemCount(): Int {
        return contentList.size
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        val item = contentList[position]
        holder.text.text = item
        holder.itemView.setOnClickListener {
            var intent = Intent(mContext, Display::class.java)
            intent.putExtra("msg", item)
            mContext.startActivity(intent)
            //mContext.startActivity(intent, ActivityOptions.makeSceneTransitionAnimation(mContext as Activity?).toBundle())
        }
    }

//    public fun setFilter(filterList: List<String?>, keyWord :String) {
//        keyWord = filterList
//        notifyDataSetChanged()
//    }
}